src/configuration.ts

```diff
@@ -24,6 +24,9 @@ import {arrayify, objectify, stripComments} from "./utils";
 
 export interface IConfigurationFile {
     extends?: string | string[];
+    linterOptions?: {
+        typeCheck?: boolean,
+    };
     rulesDirectory?: string | string[];
     rules?: any;
 }
```

Comment:
`extendConfigurationFile` further down in this file should probably support this field